ibm/service/cos/resource_ibm_cos_backup_policy_test.go

```diff
@@ -0,0 +1,919 @@
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { acc.TestAccPreCheck(t) },
+		Providers:    acc.TestAccProviders,
+		CheckDestroy: testAccCheckIBMCosBucketDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccCheckIBMCosBackup_Policy_Non_Versioning_Source_Bucket_Invalid(acc.CosCRN, bucketName, backupVaultName, region, accountID, policyName, guid),
+				ExpectError: regexp.MustCompile("Source bucket does not have versioning enabled"),
+			},
+		},
+	})
+}
+
+func TestAccIBMCosBackup_Policy_Multiple_Invalid(t *testing.T) {
+	accountID := acc.IBM_AccountID_REPL
+	bucketName := fmt.Sprintf("terraform-backup-source%d", acctest.RandIntRange(10, 100))
+	backupVaultName := fmt.Sprintf("terraform-backup-vault%d", acctest.RandIntRange(10, 100))
+	region := "us"
+	guid := strings.Split(acc.CosCRN, ":")[7]
+	policyName1 := fmt.Sprintf("backup-policy1%d", acctest.RandIntRange(10, 100))
+	policyName2 := fmt.Sprintf("backup-policy2%d", acctest.RandIntRange(10, 100))
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { acc.TestAccPreCheck(t) },
+		Providers:    acc.TestAccProviders,
+		CheckDestroy: testAccCheckIBMCosBucketDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccCheckIBMCosBackup_Policy_Multiple_Invalid(acc.CosCRN, bucketName, backupVaultName, region, accountID, policyName1, policyName2, guid),
+				ExpectError: regexp.MustCompile("Bad Request"),
+			},
+		},
+	})
+}
+
+func TestAccIBMCosBackup_Policy_Invalid_Backup_Type(t *testing.T) {
+	accountID := acc.IBM_AccountID_REPL
+	bucketName := fmt.Sprintf("terraform-backup-source%d", acctest.RandIntRange(10, 100))
+	backupVaultName := fmt.Sprintf("terraform-backup-vault%d", acctest.RandIntRange(10, 100))
+	region := "us"
+	guid := strings.Split(acc.CosCRN, ":")[7]
+	policyName := fmt.Sprintf("backup-policy%d", acctest.RandIntRange(10, 100))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { acc.TestAccPreCheck(t) },
+		Providers:    acc.TestAccProviders,
+		CheckDestroy: testAccCheckIBMCosBucketDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccCheckIBMCosBackup_Policy_Invalid_Backup_Type(acc.CosCRN, bucketName, backupVaultName, region, accountID, policyName, guid),
+				ExpectError: regexp.MustCompile("Error:"),
+			},
+		},
+	})
+}
+
+func testAccCheckIBMCosBackup_Policy_Basic_Valid(instance_id string, bucketName string, bucketVaultName string, region string, accountId string, policyName string, guid string) string {
+
+	return fmt.Sprintf(`
+
+	resource "ibm_cos_bucket" "bucket" {
+		bucket_name           = "%s"
+		resource_instance_id  = "%s"
+		cross_region_location =  "us"
+		storage_class          = "standard"
+		object_versioning {
+		  enable  = true
+		}
+	  
+	  }
+	resource "ibm_cos_backup_vault" "backup-vault" {
+		backup_vault_name           = "%s"
+		service_instance_id  = "%s"
+		region  = "%s"
+	}
+
+resource "ibm_iam_authorization_policy" "policy" {
```

Comment:
indentation issue

Reply:
This applies to all the instances in this file. Maybe in vscode you are not seeing this, but here it is clearly visible

Reply:
done